### PR TITLE
gui: add env var to override log_level

### LINKED ISSUE #782: GUI: Installer: make it possible to change the log level
It's pretty irritating to have to recompile when debugging the GUI installer, since the log level can't be change in some other way.

### PR DESCRIPTION
**Pull request by edouardparis**
close #782

### CODE REVIEW

File: gui/src/main.rs [hunk 1]
```diff
@@ -98,7 +98,10 @@ impl Application for GUI {
     }
 
     fn new(config: Config) -> (GUI, Command<Self::Message>) {
-        let logger = Logger::setup(LevelFilter::INFO);
+        let log_level = std::env::var("LOG_LEVEL")
+            .ok()
+            .and_then(|s| LevelFilter::from_str(&s).ok());
```
Comment (darosior) [inline gui/src/main.rs:103]: Next time let's not ignore errors, we can print it or something but silently ignoring it is maddening when debugging.

File: gui/README.md [hunk 2]
```diff
@@ -30,6 +30,8 @@ will be spawned that will guide you in the processing of configuring Liana.
 If the software is started and a reachable `lianad` is running, it will plug to it via `lianad`'s
 JSONRPC interface.
 
+The environment variable `LOG_LEVEL` with values `info`, `debug`, `trace`, overrides the current log settings.
```
[the remark this answers is not in the document]
Comment (darosior) [reply gui/README.md:33]: Hmm it appears to only be for the installer. At least i tried and still have my config value take precedence for the wallet side of things (which is good, just need to update the README).

**Review by darosior — COMMENTED** [review]
It appears to work for some logs but not all. For instance when setting LOG_LEVEL=error (vs LOG_LEVEL=trace) i stop getting the trace-level message from the graphic libraries, but i still get some INFO messages:
```
$ LOG_LEVEL=error cargo run 
    Finished dev [unoptimized + debuginfo] target(s) in 0.23s
     Running `target/debug/liana-gui`
  2023-12-08T09:15:53.324482Z  INFO liana_gui::installer::step::bitcoind:236: Writing to file /home/darosior/.liana/bitcoind/datadir/bitcoin.conf

  2023-12-08T09:15:53.341031Z  INFO liana_gui::installer::step::bitcoind:732: Downloading bitcoind version 25.1...
```

**Comment by darosior** [comment]
My point is you shouldn't get INFO message if you set the level to ERROR. That's was the point of my testing.

**Comment by edouardparis** [comment]
Fixed and rebased, note that we want LOG_LEVEL value to override if present, the gui settings log_level field.

**Pull request merged**